yarpl/include/yarpl/flowable/EmitterFlowable.h

```diff
@@ -164,8 +161,12 @@ class EmiterSubscription : private Subscription, private Subscriber<T> {
       int64_t emitted;
       bool done;
 
-      std::tie(emitted, done) =
-          emiter_->emit(get_ref<Subscriber<T>>(this), current);
+      std::tie(emitted, done) = emiter_->emit(this_subscriber, current);
+
+      if(this->count() == 1) {
+        assert(requested_.load(std::memory_order_relaxed) == kCanceled);
```

Comment:
Can these be `DCHECK_EQ` and `DCHECK`?  I dislike using `assert` directly because it tends to result in unused variables and broken opt builds.